### PR TITLE
Prescribe temperature field

### PR DESCRIPTION
**Pull request by gassmoeller**
As the changelog entry says this adds the option to 'prescribe' the temperature from the material model instead of solving the temperature equation. I have a model with a different way of computing temperature, but would still like to solve for compositions that is why I need to individually switch off temperature. There might be some test failures, because the output of `PrescribedFieldOutputs` now contains an additional entry for temperature, but no solution should have changed. I added a test case modelled after the existing one for composition.

### CODE REVIEW

File: source/simulator/parameters.cc [hunk 1]
```diff
@@ -1265,6 +1287,18 @@ namespace aspect
         AssertThrow (false, ExcNotImplemented());
     }
 
+    {
```
Comment (jdannberg) [inline source/simulator/parameters.cc]: Can you put this into a separate subsection ("Temperature field"?), and then move it to above or below where the same is done for composition?

File: source/simulator/solver_schemes.cc [hunk 2]
```diff
@@ -188,7 +211,7 @@ namespace aspect
               // outputs into the prescribed field before we assemle and solve the equation
```
Comment (jdannberg) [inline source/simulator/solver_schemes.cc]:
```suggestion
              // outputs into the prescribed field before we assemble and solve the equation
```

File: tests/prescribed_field_temperature.prm [hunk 3]
```diff
@@ -0,0 +1,86 @@
+# A testcase that demonstrates that interpolating material model
+# outputs into compositional fields works.
```
Comment (jdannberg) [inline tests/prescribed_field_temperature.prm]:
```suggestion
# outputs into the temperature field works.
```

File: include/aspect/simulator.h [hunk 4]
```diff
@@ -1255,23 +1255,23 @@ namespace aspect
 
 
       /**
-       * Interpolate material model outputs onto a compositional field. For the field
-       * whose index is given in the @p compositional_index, this function
+       * Interpolate material model outputs onto a advection field (temperature
+       * or composition). For the field indicated by @p adv_field, this function
```
[the remark this answers is not in the document]
Comment (gassmoeller) [reply include/aspect/simulator.h]: But `adv_field` is not an index, it is a struct of type AdvectionField that contains much more than the index. I changed it to: `For the field identified by the AdvectionField @p adv_field,`

**Comment by gassmoeller** [comment]
I addressed all comments. Lets see what the tester says.

**Pull request merged**